app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -97,6 +98,10 @@ open class HomeActivity : AppCompatActivity() {
 
         setupToolbarAndNavigation()
 
+        if (intent.getBooleanExtra(EXTRA_FINISH_ONBOARDING, false)) {
+            FenixOnboarding(this).finish()
```

Comment:
You are right, but the good news is that all of `FenixOnboarding`'s state is recreated from `SharedPreferences` every time it is instantiated. There is no runtime state passed between instances. So, this is basically a convenience method for setting the onboarding status in the app's shared preferences. 

I agree 100% that it would be good to move to `Components` so that we do not have to instantiate this multiple times (here [potentially], HomeFragment and other places that might be added in the future).  Each instantiation requires reading from `SharedPreferences`, an expensive operation. Great idea and great catch!